Refactor tutorial index (#2767)
pytorch/audio: docs/source/index.rst

   kaldi_io
   utils

.. toctree::
   :maxdepth: 1
   :caption: Prototype API Reference
   :hidden:

   prototype
   prototype.functional
   prototype.models
   prototype.pipelines

.. toctree::
   :maxdepth: 1
   :caption: PyTorch Libraries
   :hidden:

   PyTorch <https://pytorch.org/docs>
   torchaudio <https://pytorch.org/audio>
   torchtext <https://pytorch.org/text>
   torchvision <https://pytorch.org/vision>
   TorchElastic <https://pytorch.org/elastic/>
   TorchServe <https://pytorch.org/serve>
   PyTorch on XLA Devices <http://pytorch.org/xla/>

Tutorials
---------

.. customcardstart::

.. customcarditem::
   :header: Loading waveform Tensors from files and saving them

>    :card_description: Learn how to use query/load audio files and save waveform tensors to files, using <code>torchaudio.info</code>, <code>torchaudio.load</code> and <code>torchaudio.save</code> functions.

Reviewer: ```suggestion
   :card_description: Learn how to query/load audio files and save waveform tensors to files, using <code>torchaudio.info</code>, <code>torchaudio.load</code> and <code>torchaudio.save</code> functions.
```

   :image: _images/sphx_glr_audio_io_tutorial_001.png
   :link: tutorials/audio_io_tutorial.html
   :tags: I/O

.. customcarditem::
   :header: Streaming media decoding with StreamReader
   :card_description: Learn how to load audio/video to Tensors using <code>torchaudio.io.StreamReader</code> class.
   :image: _images/sphx_glr_streamreader_basic_tutorial_001.png
   :link: tutorials/streamreader_basic_tutorial.html
   :tags: I/O,StreamReader
   
.. customcarditem::
   :header: Advanced usage of StreamReader

>    :card_description: Learn how to load media from hardware devices, generate synthetic audio/video, and filters.

Reviewer: ```suggestion
   :card_description: Learn how to load media from hardware devices, generate synthetic audio/video, and filter.
```

   :image: /Users/moto/Development/torchaudio/docs/streamreader.gif
   :link: tutorials/streamreader_advanced_tutorial.html
   :tags: I/O,StreamReader

.. customcarditem::
   :header: Streaming media encoding with StreamWriter
   :card_description: Learn how to save audio/video with <code>torchaudio.io.StreamWriter</code>
   :image: /Users/moto/Development/torchaudio/docs/streamwriter.gif
   :link: tutorials/streamwriter_basic_tutorial.html
   :tags: I/O,StreamWriter
   
.. customcarditem::
   :header: Playing media with StreamWriter
   :card_description: Learn how to play audio/video with <code>torchaudio.io.StreamWriter</code>
   :image: /Users/moto/Development/torchaudio/docs/udp.gif
   :link: tutorials/streamwriter_advanced_tutorial.html
   :tags: I/O,StreamWriter

.. customcarditem::
   :header: Accelerated video I/O with NVDEC/NVENC

>    :card_description: Learn how to setup HW acceleratoin for video processing

Reviewer: ```suggestion
   :card_description: Learn how to setup HW acceleration for video processing
```

   :image: _images/hw_acceleration_tutorial_68_1.png
   :link: hw_acceleration_tutorial.html
   :tags: I/O,StreamReader,StreamWriter

.. customcarditem::
   :header: Audio resampling with bandlimited sinc interpolation  
   :card_description: Learn how to resample audio tensor
   :image: _images/sphx_glr_audio_resampling_tutorial_001.png
   :link: tutorials/audio_resampling_tutorial.html
   :tags: Preprocessing

.. customcarditem::
   :header: Audio data augmentation
   :card_description: Learn how to use TorchAudio's functional/transforms modules to perform data augmentation
   :image: _images/sphx_glr_audio_data_augmentation_tutorial_007.png
   :link: tutorials/audio_data_augmentation_tutorial.html
   :tags: Preprocessing

.. customcarditem::
   :header: Audio feature extraction
   :card_description: Learn how to use TorchAudio's functional/transforms modules to extract features from waveform
   :image: _images/sphx_glr_audio_feature_extractions_tutorial_011.png
   :link: tutorials/audio_feature_extractions_tutorial.html
   :tags: Preprocessing

.. customcarditem::
   :header: Audio feature augmentation

>    :card_description: Learn how to use TorchAudio's functional/transforms modules to peform feature augmentation

Reviewer: ```suggestion
   :card_description: Learn how to use TorchAudio's functional/transforms modules to perform feature augmentation
```

   :image: _images/sphx_glr_audio_feature_augmentation_tutorial_002.png
   :link: tutorials/audio_feature_augmentations_tutorial.html
   :tags: Preprocessing

.. customcarditem::
   :header: Audio dataset
   :card_description: Learn how to use TorchAudio's dataset module
   :image: _images/sphx_glr_audio_datasets_tutorial_001.png
   :link: tutorials/audio_datasets_tutorial.html
   :tags: Dataset

.. customcarditem::
   :header: AM inference with Wav2Vec2

>    :card_description: Learn how to acoustice model inference with Wav2Vec2

Reviewer: ```suggestion
   :card_description: Learn how to perform acoustic model inference with Wav2Vec2
```

   :image: _images/sphx_glr_speech_recognition_pipeline_tutorial_002.png
   :link: tutorials/speech_recognition_pipeline_tutorial.html
   :tags: ASR,wav2vec2

.. customcarditem::
   :header: LM inference with CTC Beam Seach Decoder

>    :card_description: Learn how to lexicon, language model and beam search decoder in ASR

Reviewer: ```suggestion
   :card_description: Learn how to perform ASR beam search decoding with lexicon and language model
```

   :image: _images/sphx_glr_asr_inference_with_ctc_decoder_tutorial_001.png
   :link: tutorials/asr_inference_with_ctc_decoder_tutorial.html

>    :tags: Pipelines,ASR,wav2vec2,CTC-Decoder,KenLM

Reviewer: I don't think KenLM should be a subsection

```suggestion
   :tags: Pipelines,ASR,wav2vec2,CTC-Decoder
```

.. customcarditem::
   :header: Online ASR with Emformer RNN-T
   :card_description: Learn how to perform online ASR with RNN-T and <code>torchaudio.io.StreamReader</code>.
   :image: https://download.pytorch.org/torchaudio/tutorial-assets/emformer_rnnt_streamer_context.png
   :link: tutorials/online_asr_tutorial.html
   :tags: Pipelines,ASR,RNNT,StreamReader
   
.. customcarditem::

>    :header: Readl-time microphone ASR with Emformer RNN-T

Reviewer: ```suggestion
   :header: Real-time microphone ASR with Emformer RNN-T
```

>    :card_description: Learn how to transcribe speech fomr micrphone with RNN-T and <code>torchaudio.io.StreamReader</code>.

Reviewer: ```suggestion
   :card_description: Learn how to transcribe speech fomr microphone with RNN-T and <code>torchaudio.io.StreamReader</code>.
```

   :image: https://3.bp.blogspot.com/-aw5KRAWmnKE/WK7fGITeEqI/AAAAAAABCA4/j5G8qbugxmkdDWhmlAXy7ZYbTYhaaIv1ACLcB/s800/microphone_mark.png
   :link: tutorials/device_asr.html
   :tags: Pipelines,ASR,RNNT,StreamReader

.. customcarditem::
   :header: Forced Alignment with Wav2Vec2
   :card_description: Learn how to align text to speech with Wav2Vec 2.0
   :image: _images/sphx_glr_forced_alignment_tutorial_005.png
   :link: tutorials/forced_alignment_tutorial.html
   :tags: Pipelines,Forced-Alignment,wav2vec2

.. customcarditem::
   :header: Text-to-Speech with Tacotron2
   :card_description: Learn how to generate speech from text with Tacotron2
   :image: _images/sphx_glr_tacotron2_pipeline_tutorial_003.png
   :link: tutorials/tacotron2_pipeline_tutorial.html
   :tags: Pipelines,TTS-(Text-to-Speech)
   
.. customcarditem::
   :header: Speech Enhancement with MVDR Beamforming
   :card_description: Learn how to improve speech quality with MVDR Beamforming
   :image: _images/sphx_glr_mvdr_tutorial_001.png
   :link: tutorials/mvdr_tutorial.html
   :tags: Pipelines,Speech-Enhancement

.. customcarditem::
   :header: Music Source Separation with Hybrid Demucs

>    :card_description: Learn how to perform music separation with pre-trained Hybrid Demucs

Reviewer: ```suggestion
   :card_description: Learn how to perform music source separation with pre-trained Hybrid Demucs
```
